src/interfaces/coral_web/src/pages/c/[id]/index.tsx

```diff
@@ -40,12 +41,19 @@ const ConversationPage: NextPage<Props> = () => {
     ? router.query.id[0]
     : (router.query.id as string);
 
-  const {
-    data: conversation,
-    isLoading,
-    isError,
-    error,
-  } = useConversation({ conversationId: urlConversationId });
+  const { data, isLoading, isError, error } = useConversation({
+    conversationId: urlConversationId,
+  });
+  const conversation = useMemo(() => {
+    if (!data) return;
+
+    data.messages = data.messages.map((message) => ({
+      ...message,
+      text: replaceCodeBlockWithIframe(message.text),
```

Comment:
would it make sense to move this to `mapHistoryToMessages`? since we do mapping for `replaceTextWithCitations` there too